src/python/dxpy/asset_builder.py

```diff
@@ -202,7 +201,28 @@ def build_asset(args):
         if not args.json:
             print("Uploading input files for the AssetBuilder", file=sys.stderr)
 
-        conf_file = dx_upload(asset_conf_file, dest_project_name, dest_folder_name, args.json)
+        # hide nextflow & nextaur asset records
+        if (str(asset_conf['name']) in ["nextflow", "nextaur"]):
+            asset_conf["hidden"] = True
+        # override visibility using runtime setting
+        elif args.visibility:
+            asset_conf["hidden"] = (args.visibility == 'hidden')
+        
+        # set visibility to false if not been specified yet
+        if not asset_conf.get("hidden"):
+            asset_conf["hidden"] = False
+
+        with tempfile.NamedTemporaryFile(dir=args.src_dir, suffix=".json", mode='w+') as temp_asset_conf:
```

Comment:
Just want to make sure the user's dxasset.json is intact. Any suggestions here? @commandlinegirl

Reply:
Do we need to update the file at all? 

Wouldn't it make sense to make it hidden only if dxasset.json contains "hidden" field? Thus we would not introduce any new argument and we would never forget to make an asset hidden.

Reply:
it is requested as in https://jira.internal.dnanexus.com/browse/APPS-1565
`expose "--visibility hidden" as an option to asset_builder so all future asset building can have the option to hide the asset.` so both runtime and dxasset.json are considered.

Reply:
Marek's comment makes sense to me. We could also completely ditch `--visibility hidden` option and only have the new dxasset.json "visibility" field. I'm not sure if `dx_upload()` has the visibility parameter but maybe we can add it (as an optional).

Reply:
I think for the sake of time we can only allow setting the option from the asset.json, if it make it easier and faster for us and add the commandline option separately.

Reply:
> Wouldn't it make sense to make it hidden only if dxasset.json contains "hidden" field? Thus we would not introduce any new argument and we would never forget to make an asset hidden (as it would be in the dxasset.json).

I already include the case when no runtime spec is given:
https://github.com/dnanexus/dx-toolkit/blob/4b86fe0a10108168908c0e30a6d7d64d31fe3baf/src/python/dxpy/asset_builder.py#L204-L213

Reply:
Yes, we can remove it. This will simplify things. Thank you @YuxinShi0423!